### ISSUE TITLE
Gemini

### ISSUE THREAD

**Issue by HarolddeBruijn** [aittalam/PicoGopher#1]
I wonder if it would be feasable at all to extend this into the gemini protocol.

**Comment by aittalam** [comment]
Yes, it should be possible! [This](https://github.com/Astrrra/ESP8266GeminiServer) is a project that does this on an ESP8266, and [this](https://github.com/w84death/ESP8266GeminiServer) is a fork that I saw running (i.e. I connected to the server while it was running, powered by a solar panel :-).

I think extending picogopher to do this should be mostly a matter of refactoring the code to work with the rest. I wanted to do it myself but I did not find the time - would be glad to see it running though, and if you want to work on this I'd be happy to lend a hand!

**Comment by aittalam** [comment]
I recently added [this branch](https://github.com/aittalam/PicoGopher/tree/mala/adding_gemini) with an early implementation of a gemini server. This is heavily inspired by Solderpunk's [GeGoBi](https://git.sr.ht/~solderpunk/gegobi), and in theory it supports both capsules and gopherholes, serving both of them with the gemini protocol.
I have not tested enough to already merge it into main, but it will likely be my next main update - feel free to play already with it and lmk how it goes :-)
